src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/AddressServiceImpl.java

```diff
@@ -90,7 +92,23 @@ public AddressIndexAddressCompositeDTO uprnQuery(long uprn) throws CTPException
             addressResult.getStatus().getCode(),
             addressResult.getStatus().getMessage());
       }
+
       AddressIndexAddressCompositeDTO address = addressResult.getResponse().getAddress();
+
+      // Validate address type
+      try {
+        AddressType.valueOf(address.getCensusAddressType());
```

Comment:
Asserting the addresses type here and bitching will mean that a future use case that needs to fetch an address by uprn where addressType NA WOULD be acceptable will fail. And probably not until that new use case was exercised such that it found an NA address in whitelodge.

The rejection of an NA typed address should live higher up the stack, nearer the use case that does need to make this assertion about the type. Suggest moving to CaseServiceImpl.createNewCachedCase()

Reply:
CensusAddressType validation moved into CaseServiceImpl which does this validation just after rejecting Scottish addresses.
Unit tests updated to correspond with new location of this validation.